Sources/MongoSwift/MongoDatabase.swift

```diff
@@ -94,20 +94,24 @@ public struct CreateCollectionOptions: Encodable, CodingStrategyProvider {
     /// retrieve the collection using `MongoDatabase.collection`.
     public let writeConcern: WriteConcern?
 
+    // swiftlint:disable redundant_optional_initialization
+    // to get synthesized decodable conformance for the struct, these strategies need default values.
+
     /// Specifies the `DateCodingStrategy` to use for BSON encoding/decoding operations performed by this collection.
     /// It is the responsibility of the user to ensure that any `Date`s already stored in this collection can be
     /// decoded using this strategy.
-    public let dateCodingStrategy: DateCodingStrategy?
+    public var dateCodingStrategy: DateCodingStrategy? = nil
```

Comment:
giving these default values (for the reason I noted by the "disable" above) also meant they needed to be vars so they can be overwritten. I figured that was OK as all options properties are going to become `var`s soon anyway as of [SWIFT-355](https://jira.mongodb.org/browse/SWIFT-355) which is part of the intern plan.